resources/scripts/dcs_liberation.lua

```diff
@@ -39,46 +39,54 @@ write_state = function()
     -- messageAll("Done writing DCS Liberation state.")
 end
 
-debriefing_file_location = nil
-if dcsLiberation then 
-    debriefing_file_location = dcsLiberation.installPath
-end
-if debriefing_file_location then
-    logger:info("Using DCS Liberation install folder for state.json")
-else
-    if os then
-        debriefing_file_location = os.getenv("LIBERATION_EXPORT_DIR")
-        if debriefing_file_location then debriefing_file_location = debriefing_file_location .. "\\" end
-    end
-    if debriefing_file_location then
-        logger:info("Using LIBERATION_EXPORT_DIR environment variable for state.json")
-    else
-        if os then
-            debriefing_file_location = os.getenv("TEMP")
-            if debriefing_file_location then debriefing_file_location = debriefing_file_location .. "\\" end
+
+local function discoverDebriefingFilePath()
+    local function insertFileName(directoryOrFilePath, overrideFileName)
+        if overrideFileName then
+            logger:info("Using LIBERATION_EXPORT_STAMPED_STATE to locate the state.json")
+            return directoryOrFilePath .. os.time() .. "-state.json"
         end
-        if debriefing_file_location then
-            logger:info("Using TEMP environment variable for state.json")
-        else
-            if lfs then
-                debriefing_file_location = lfs.writedir()
-            end
-            if debriefing_file_location then
-                logger:info("Using DCS working directory for state.json")
-            end
+
+        local filename = "state.json"
+        if not (directoryOrFilePath:sub(-#filename) == filename) then
+            return directoryOrFilePath .. filename
         end
+
+        return directoryOrFilePath
     end
-end
-if debriefing_file_location then
-    local filename = "state.json"
-    if not debriefing_file_location:sub(-#filename) == filename then
```

Comment:
It is noteworthy that this line would have never worked. This has been corrected with bracing at the `not` would operate on the `debriefing_file_location:sub(-#filename)` statement, and not on the expanded binary statement of `debriefing_file_location:sub(-#filename) == filename`

E.g. Original code is interpretted as
```
-- original
if (not debriefing_file_location:sub(-#filename)) == filename then
-- ^ --- note this bracket for emphasis ------- ^

-- corrected in PR
if not (debriefing_file_location:sub(-#filename) == filename) then
--     ^ --- note this bracket for emphasis --------------- ^
```